pkg/csi/cinder/nodeserver.go

```diff
@@ -157,16 +255,33 @@ func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	if len(volumeID) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "NodeUnpublishVolume volumeID must be provided")
 	}
-	_, err := ns.Cloud.GetVolume(volumeID)
-	if err != nil {
-		if cpoerrors.IsNotFound(err) {
-			return nil, status.Error(codes.NotFound, "Volume not found")
+
+	//TODO: ephemeralVolume := req.GetVolumeContext()["csi.storage.k8s.io/ephemeral"] == "true"
+	mode := ns.Driver.mode
+	var vol volumes.Volume
+	if mode == "ephemeral" {
+		volName := fmt.Sprintf("ephemeral-%s", volumeID)
+		vols, err := ns.Cloud.GetVolumesByName(volName)
+		if err != nil {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("GetVolume failed with error %v", err))
 		}
-		return nil, status.Error(codes.Internal, fmt.Sprintf("GetVolume failed with error %v", err))
+		if len(vols) > 0 {
```

Comment:
Is this correct? If we have 2 volumes with same name , dont we need to throw error ? l

Reply:
i can include a check, but it is not possible for ephemeral case,  as volume name includes unique id generated by kubernetes